actionpack/lib/abstract_controller/base.rb

```diff
@@ -11,6 +11,24 @@ module AbstractController
   class ActionNotFound < StandardError
   end
 
+  # Raised when a nested respond_to is triggered and the content types of each
+  # are incompatible. For exampe:
+  #
+  #  respond_to do |outer_type|
+  #    outer_type.js do
+  #      respond_to do |inner_type|
+  #        inner_type.html { render body: "HTML" }
+  #      end
+  #    end
+  #  end
+  class RespondToMismatchError < StandardError
```

Comment:
Should this error class be moved to a different file? Maybe moving it to `actionpack/lib/action_controller/metal/mime_responds.rb` would better separate concerns.

Reply:
I ended up searching for where other similar exceptions were defined (ex. `ActionController::UnknownFormat`) and stumbled upon `actionpack/lib/action_controller/metal/exceptions.rb`. So, I placed the exception there.